File: src/Level1/K번째수.java
package Level1;
import java.util.Arrays;

public class K번째수 {
    public int[] solution(int[] array, int[][] commands) {
    int[] answer = new int[commands.length];

    for(int i = 0 ; i < commands.length; i++){
        int firstIndex = commands[i][0];
        int lastIndex = commands[i][1];
        int[] cutArray = new int[lastIndex-firstIndex+1];
        int k = 0;
        for(int j = firstIndex-1; j < lastIndex; j++){
            cutArray[k] = array[j];
            k++;
        }
        Arrays.sort(cutArray);
        answer[i] = cutArray[commands[i][2]-1];
    }
    return answer;
}
}

File: src/Level1/K번째수.kt
package Level1

fun main(){
    solution(intArrayOf(1, 5, 2, 6, 3, 7, 4), arrayOf(intArrayOf(2, 5, 3), intArrayOf(4, 4, 1), intArrayOf(1, 7, 3))).forEach {
        println(it)
    }
}

fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
    var answer = IntArray(commands.size, {0})
    var i = 0
    for(command in commands){
        val sortedArray = array.sliceArray(IntRange(command[0] - 1, command[1] - 1))
        sortedArray.sort()
        answer[i] = sortedArray[command[2] - 1]
        i++
    }
    return answer
}